packages/core/src/extensions/SlashMenu/defaultSlashMenuItems.ts

```diff
@@ -0,0 +1,122 @@
+import { BlockNoteEditor } from "../../BlockNoteEditor";
+import { BlockSchema, PartialBlock } from "../Blocks/api/blockTypes";
+import {
+  BaseSlashMenuItem,
+  createBaseSlashMenuItem,
+} from "./BaseSlashMenuItem";
+
+function insertOrUpdateBlock<BSchema extends BlockSchema>(
+  editor: BlockNoteEditor<BSchema>,
+  block: PartialBlock<BSchema>
+) {
+  const currentBlock = editor.getTextCursorPosition().block;
+
+  if (
+    (currentBlock.content.length === 1 &&
+      currentBlock.content[0].type === "text" &&
+      currentBlock.content[0].text === "/") ||
+    currentBlock.content.length === 0
+  ) {
+    editor.updateBlock(currentBlock, block);
+  } else {
+    editor.insertBlocks([block], currentBlock, "after");
+    editor.setTextCursorPosition(editor.getTextCursorPosition().nextBlock!);
+  }
+}
+
+export const getDefaultSlashMenuItems = <BSchema extends BlockSchema>(
+  schema: BSchema
+) => {
+  const slashMenuItems: BaseSlashMenuItem<BSchema>[] = [];
+
+  if ("heading" in schema && "level" in schema.heading.propSchema) {
+    if (schema.heading.propSchema.level.values?.includes("1")) {
+      slashMenuItems.push(
+        // Command for creating a level 1 heading
+        createBaseSlashMenuItem(
```

Comment:
I don't think this needs to be a function, we can get rid of `createBaseSlashMenuItem` all together I think?